test/base/test_mps.jl

```diff
@@ -755,6 +755,18 @@ end
     truncate!(M; site_range=3:7, maxdim=2)
     @test linkdims(M) == [2, 4, 2, 2, 2, 2, 8, 4, 2]
   end
+
+  @testset "truncate! with truncation_errors" begin
+    N = 10
+    nbonds = N - 1
+    M = basicRandomMPS(10; dim=10)
+    truncation_errors = Ref{Vector{Float64}}()
+    truncation_errors[] = fill(-1.0, nbonds) # set to something other than zero for test.
```

Comment:
It would be better to design the code so that users don't have to initialize the list ahead of time. We could add a line:
```julia
if isnothing(truncation_errors)
  truncation_errors[] = Vector{real(scalartype(M))}(undef, length(M) - 1)
end
```
at the top of `truncate!(...)`.